### PR TITLE
Add v1alpha2 embedded Target Allocator struct

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Add and use the embedded Target Allocator struct for v1alpha2. The changes mostly involve label selectors for Prometheus CR.

Worth noting that this involves a change in semantics for said selectors, as an empty selector now selects nothing, which is in line with the standard, but different from v1alpha1.

### CODE REVIEW

File: apis/v1alpha2/allocation_strategy.go [hunk 1]
```diff
@@ -31,5 +31,5 @@ const (
 	TargetAllocatorAllocationStrategyConsistentHashing TargetAllocatorAllocationStrategy = "consistent-hashing"
 
 	// TargetAllocatorFilterStrategyRelabelConfig targets will be consistently drops targets based on the relabel_config.
-	TargetAllocatorFilterStrategyRelabelConfig TargetAllocatorFilterStrategy = "consistent-hashing"
+	TargetAllocatorFilterStrategyRelabelConfig TargetAllocatorFilterStrategy = "relabel-config"
```
Comment (swiatekm-sumo) [inline apis/v1alpha2/allocation_strategy.go:34]: This is a small bug fix, the value was simply wrong.

File: apis/v1alpha2/opentelemetrycollector_types.go [hunk 2]
```diff
@@ -165,6 +166,84 @@ type OpenTelemetryCollectorSpec struct {
 	DeploymentUpdateStrategy appsv1.DeploymentStrategy `json:"deploymentUpdateStrategy,omitempty"`
 }
 
+// TargetAllocatorEmbedded defines the configuration for the Prometheus target allocator, embedded in the
+// OpenTelemetryCollector spec.
+type TargetAllocatorEmbedded struct {
```
Comment (jaronoff97) [inline apis/v1alpha2/opentelemetrycollector_types.go:171]: should we have this use the common fields?

**Review by jaronoff97 — COMMENTED** [review]
one question, otherwise looking good!

File: apis/v1alpha2/opentelemetrycollector_types.go (hunk 2 above)
Comment (swiatekm-sumo) [reply apis/v1alpha2/opentelemetrycollector_types.go:171]: I'd rather do that in the TargetAllocator CR proper, and slowly deprecate most of the common fields in this embedded struct. For the migration, I'd like to keep them the same for simplicity.

**Review by jaronoff97 — APPROVED** [review]
The semantics change is fine to me IMO, when we actually start using this in #2565 we should include this in a changelog maybe.

**Pull request merged**